### PR TITLE
Fix #481: URLEncode UI resources

### LINKED ISSUE #481: Static Resource Paths Aren't URL Encoded In Route Handler
### Describe the bug

When serving assets with spaces in their name, they get redirected back to the SPA index.html page instead of being served up. This appears to be because the assets in the browser are URL encoded and the static handler is not URL encoding the file names and therefore saying it "doesn't handle it".

Example: URL for an asset is http://orchestrator.default.127.0.0.1.sslip.io:8080/assets/Destiny%20Logo.png, gets redirected back to index.html
Example: URL for an asset (no space), http://orchestrator.default.127.0.0.1.sslip.io:8080/assets/destinylogo.png gets served up just fine.

Quinoa recognizes the asset:

Quinoa is scanning directory: /linux-repos/orchestrator/src/main/webui/src
Quinoa is scanning directory: /linux-repos/orchestrator/src/main/webui/src/assets
Quinoa is watching: /linux-repos/orchestrator/src/main/webui/src/assets/Destiny Logo.png
Quinoa is watching: /linux-repos/orchestrator/src/main/webui/src/assets/destinylogo.png

Quinoa copies it to the build directory just fine:

image

### Quinoa version

2.0.8

### Quarkus version

3.2.2.Final

### Build / Runtime

Angular

### Package Manager

NPM

### Steps to reproduce the behavior

1. Create a Quinoa Angular app and put a static asset with spaces in the name in src/main/webui/src/assets/
2. Create an angular app which serves up the image, e.g. `<h1 class="mat-h1">Destiny Logo</h1>`
3. Run `./gradlew quarkusDev`, notice that it works just fine because `ng serve` is watching the files with URL encoded file names for the route handler
4. Run `./gradlew clean quarkusBuild -Dquarkus.knative.deploy=true` and go to the deployed instance, notice that the file isn't displayed due to a redirection back to index.html
5. Change file to not have spaces and repeat 3 & 4 but this time Step 4 shows the image.

### Expected behavior

Static assets with spaces in their name are served up properly.

### PR DESCRIPTION
**Pull request by melloware**
Fix #481: URLEncode UI resources

Basically we need to store the UI Resource names at rest as URI Encoded so they match what the browser is sending.

### CODE REVIEW

**Comment by melloware** [comment]
I have no idea why the Integration tests are failing.

**Comment by shivam-sharma7** [comment]
> I have no idea why the Integration tests are failing.

did you test locally?

**Comment by shivam-sharma7** [comment]
I have reverted my #486. now let's see

**Review by shivam-sharma7 — APPROVED** [review]
LGTM

File: runtime/src/main/java/io/quarkiverse/quinoa/QuinoaUIResourceHandler.java [hunk 1]
```diff
@@ -63,4 +69,30 @@ private static Handler<RoutingContext> createStaticHandler(QuinoaHandlerConfig c
...
+     * <p>
+     * URI-encode the given string using UTF-8. URIs (paths and filenames) have different encoding rules as compared to
+     * URL query string parameters. {@link URLEncoder} is actually only for www (HTML) form based query string parameter
+     * values (as used when a webbrowser submits a HTML form). URI encoding has a lot in common with URL encoding, but
+     * the space has to be %20 and some chars doesn't necessarily need to be encoded.
+     *
+     * @param string The string to be URI-encoded using UTF-8.
+     * @return The given string, URI-encoded using UTF-8, or <code>null</code> if <code>null</code> was given.
+     */
+    private static String encodeURI(String string) {
```
Comment (ia3andy) [inline runtime/src/main/java/io/quarkiverse/quinoa/QuinoaUIResourceHandler.java:84]: Can you check, I am pretty sure Vertx have something to do this?

**Comment by ia3andy** [comment]
@melloware it would be nice to have a test for this?

**Pull request merged**